core/trino-main/src/main/java/io/trino/execution/SqlTask.java

```diff
@@ -203,6 +203,16 @@ public SqlTaskIoStats getIoStats()
         return taskHolderReference.get().getIoStats();
     }
 
+    public TaskState getTaskStateSnapshot()
```

Comment:
Since other places in the code were so careful about the progression of task status I thought it might be merited, but if you don't think so I'll just rename.